sdk/storage/azure-storage-blob/azure/storage/blob/aio/_lease_async.py

```diff
@@ -3,51 +3,58 @@
 
-from .._shared.response_handlers import return_response_headers, process_storage_error
+from .._shared.response_handlers import process_storage_error, return_response_headers
 from .._serialize import get_modify_conditions
-from .._lease import BlobLeaseClient as LeaseClientBase
 
 if TYPE_CHECKING:
+    from azure.storage.blob.aio import BlobClient, ContainerClient
     from datetime import datetime
-    from .._generated.operations import BlobOperations, ContainerOperations
-    BlobClient = TypeVar("BlobClient")
-    ContainerClient = TypeVar("ContainerClient")
 
 
-class BlobLeaseClient(LeaseClientBase):
+class BlobLeaseClient(): # pylint: disable=client-accepts-api-version-keyword
     """Creates a new BlobLeaseClient.
 
     This client provides lease operations on a BlobClient or ContainerClient.
-
-    :ivar str id:
-        The ID of the lease currently being maintained. This will be `None` if no
-        lease has yet been acquired.
-    :ivar str etag:
-        The ETag of the lease currently being maintained. This will be `None` if no
-        lease has yet been acquired or modified.
-    :ivar ~datetime.datetime last_modified:
-        The last modified timestamp of the lease currently being maintained.
-        This will be `None` if no lease has yet been acquired or modified.
-
-    :param client:
+    :param  client:
         The client of the blob or container to lease.
-    :type client: ~azure.storage.blob.aio.BlobClient or
-        ~azure.storage.blob.aio.ContainerClient
-    :param str lease_id:
+    :type client: Union[BlobClient, ContainerClient]
+    :param Optional[str] lease_id:
         A string representing the lease ID of an existing lease. This value does not
         need to be specified in order to acquire a new lease, or break one.
     """
 
+    id: str
+    """The ID of the lease currently being maintained. This will be `None` if no
+    lease has yet been acquired."""
+    etag: Optional[str]
+    """The ETag of the lease currently being maintained. This will be `None` if no
+    lease has yet been acquired or modified."""
+    last_modified: Optional["datetime"]
+    """The last modified timestamp of the lease currently being maintained.
+    This will be `None` if no lease has yet been acquired or modified."""
+
+    def __init__( # pylint: disable=missing-client-constructor-parameter-credential, missing-client-constructor-parameter-kwargs
+        self, client: Union["BlobClient", "ContainerClient"],
+        lease_id: Optional[str] = None
+    ) -> None:
+        self.id = lease_id or str(uuid.uuid4())
+        self.last_modified = None
+        self.etag = None
+        if hasattr(client, 'blob_name'):
+            self._client = client._client.blob  # type: ignore # pylint: disable=protected-access
+        elif hasattr(client, 'container_name'):
+            self._client = client._client.container  # type: ignore # pylint: disable=protected-access
```

Comment:
Just seeing this not for async but applies to sync as well. Can you add a private member type here that is a union of the two generated clients. That way we can remove these type ignores:

On line 43: `_client: Union[BlobOperations, ContainerOperations]` (Be sure to import the async ones here and the sync ones in sync)

There is a chance that this causes other errors in the file (in that case, don't worry about this) but I think it will be okay because both operation types should have all the same lease operations defined.